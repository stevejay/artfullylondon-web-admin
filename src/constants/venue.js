export const VENUE_TYPE_THEATRE = 'Theatre'
export const VENUE_TYPE_ART_GALLERY = 'Art Gallery'
export const VENUE_TYPE_CONCERT_HALL = 'Concert Hall'
export const VENUE_TYPE_EXHIBITION_HALL = 'Exhibition Hall'
export const VENUE_TYPE_PERFORMING_ARTS_CENTRE = 'Performing Arts Centre'
export const VENUE_TYPE_MUSEUM = 'Museum'
export const VENUE_TYPE_CHURCH = 'Church'
export const VENUE_TYPE_CINEMA = 'Cinema'
export const VENUE_TYPE_OTHER = 'Other'

export const ALLOWED_VENUE_TYPES = [
  VENUE_TYPE_THEATRE,
  VENUE_TYPE_ART_GALLERY,
  VENUE_TYPE_CONCERT_HALL,
  VENUE_TYPE_EXHIBITION_HALL,
  VENUE_TYPE_PERFORMING_ARTS_CENTRE,
  VENUE_TYPE_MUSEUM,
  VENUE_TYPE_CHURCH,
  VENUE_TYPE_CINEMA,
  VENUE_TYPE_OTHER
]
