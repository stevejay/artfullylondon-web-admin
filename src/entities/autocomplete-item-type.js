export default {
  ENTITY: 'entity',
  LABEL: 'label'
}
