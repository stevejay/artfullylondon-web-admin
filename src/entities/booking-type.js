export default {
  NOT_REQUIRED: 'NotRequired',
  REQUIRED: 'Required',
  REQUIRED_FOR_NON_MEMBERS: 'RequiredForNonMembers'
}
