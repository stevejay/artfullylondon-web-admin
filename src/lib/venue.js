const POSTCODE_DISTRICT_REGEX = /^[A-Z]{1,2}[0-9]{1,2}/
const NEWLINES_REGEX = /\n/g

export function formatAddressForDisplay (address, postcode) {
  return address.replace(NEWLINES_REGEX, ', ') + ', ' + postcode
}

export function getPostcodeDistrict (postcode) {
  var matches = (postcode || '').match(POSTCODE_DISTRICT_REGEX)
  return matches ? matches[0] : null
}
