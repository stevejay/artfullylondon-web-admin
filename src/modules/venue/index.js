import VenueEditOrCreate from './components/venue-edit-or-create'
import VenueDetail from './components/venue-detail'
import { moduleName, reducer, selectors } from './reducers'
import sagas from './sagas'

export { VenueEditOrCreate, VenueDetail, moduleName, reducer, selectors, sagas }
