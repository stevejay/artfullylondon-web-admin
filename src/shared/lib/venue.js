const POSTCODE_DISTRICT_REGEX = /^[A-Z]{1,2}[0-9]{1,2}/

export function getPostcodeDistrict (postcode) {
  var matches = (postcode || '').match(POSTCODE_DISTRICT_REGEX)
  return matches ? matches[0] : null
}
