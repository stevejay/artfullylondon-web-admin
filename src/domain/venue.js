import _ from 'lodash'

import * as venueLib from '_src/lib/venue'
import * as timeLib from '_src/lib/time'
import { Entity } from '_src/domain/entity'
import entityType from '_src/domain/types/entity-type'
import linkType from '_src/domain/types/link-type'

const NEWLINES_REGEX = /\n/g

export class FullVenue extends Entity {
  get entityType () {
    return entityType.VENUE
  }

  getInfoBarLabel () {
    return this.venueType
  }

  get url () {
    throw new Error('url accessed')
  }

  get editUrl () {
    return `/venue/edit/${this.id}`
  }

  get pin () {
    return {
      lat: this.latitude,
      lng: this.longitude
    }
  }

  getHomepageUrl () {
    const homepage = this.getLinkByType(linkType.HOMEPAGE)
    return homepage ? homepage.url : null
  }

  createTimesDetailsOn (dateStr) {
    return timeLib.getTimesDetails(this, entityType.VENUE, dateStr)
  }

  shallowClone (newProps) {
    return new FullVenue({ ...this, ...newProps })
  }
}

export class SummaryVenue {
  constructor (entity) {
    _.extend(this, entity)
  }

  getPostcodeDistrict () {
    return venueLib.getPostcodeDistrict(this.postcode)
  }

  get entityTypeLabel () {
    return 'Venue'
  }

  get key () {
    return this.id
  }

  get url () {
    return `/venue/${this.id}`
  }

  get pin () {
    return {
      lat: this.latitude,
      lng: this.longitude
    }
  }

  get hasImage () {
    return !!this.image
  }

  createFullAddress () {
    return (
      this.address.replace(NEWLINES_REGEX, ', ') +
      (this.postcode ? ', ' + this.postcode : '')
    )
  }

  shallowClone (newProps) {
    return new SummaryVenue({ ...this, ...newProps })
  }
}
