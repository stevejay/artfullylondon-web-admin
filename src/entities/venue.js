import { ENTITY_TYPE_VENUE } from '_src/constants/entity'
import { LinkCollection } from '_src/entities/link-collection'
import * as sharedVenue from '_src/lib/venue'
import * as sharedEntity from '_src/lib/entity'
import { LINK_TYPE_HOMEPAGE } from '_src/constants/link'
import * as time from '_src/lib/time'
import * as image from '_src/lib/image'

export class SummaryVenue {
  constructor (entity) {
    if (!entity) {
      throw new Error('null or undefined entity')
    }

    this.entity = entity
    this.postcodeDistrict = sharedVenue.getPostcodeDistrict(entity.postcode)
  }

  get entityType () {
    return ENTITY_TYPE_VENUE
  }

  get entityTypeLabel () {
    return 'Venue'
  }

  get status () {
    return this.entity.status
  }

  get id () {
    return this.entity.id
  }

  get key () {
    return this.id
  }

  get name () {
    return this.entity.name
  }

  get venueType () {
    return this.entity.venueType
  }

  get postcode () {
    return this.entity.postcode
  }

  get url () {
    return sharedEntity.createEntityUrl(this.entityType, this.id)
  }

  get editUrl () {
    return sharedEntity.createEntityEditUrl(this.entityType, this.id)
  }

  get latitude () {
    return this.entity.latitude
  }

  get longitude () {
    return this.entity.longitude
  }

  get pin () {
    return {
      lat: this.latitude,
      lng: this.longitude
    }
  }

  get image () {
    return this.entity.image
  }

  get imageCopyright () {
    return this.entity.imageCopyright
  }

  get imageRatio () {
    return this.entity.imageRatio
  }

  get hasImage () {
    return !!this.image
  }

  get cardImageLoaded () {
    return !!this.entity.cardImageLoaded
  }

  get isFullEntity () {
    return false
  }

  isBeingWatched (watches) {
    return !!watches[this.id]
  }

  createWatchLabel () {
    return this.name
  }

  createWatchChangeInstruction (isBeingWatched) {
    const prefix = isBeingWatched ? 'Unwatch' : 'Watch'
    return prefix + ' this venue'
  }

  createFullAddress () {
    return sharedVenue.formatAddressForDisplay(
      this.entity.address,
      this.entity.postcode
    )
  }

  createVenuesMapIconUrl (isSelected) {
    return image.createVenueTypePngIconUrl(this.venueType, isSelected)
  }

  shallowClone (newProps) {
    const clonedEntity = Object.assign({}, this.entity, newProps || {})
    return new SummaryVenue(clonedEntity)
  }
}

export class FullVenue extends SummaryVenue {
  constructor (entity) {
    super(entity)
    this.links = new LinkCollection(entity.links)
  }

  createTimesDescriptionForDate (dateStr, timeStr, namedClosuresLookup) {
    return time.formatTimesStringForGivenDate(
      this.entity,
      dateStr,
      timeStr,
      namedClosuresLookup
    )
  }

  createTimesDetailsOn (dateStr) {
    return time.getTimesDetails(this.entity, this.entityType, dateStr)
  }

  createInfoBarLabel () {
    return this.venueType
  }

  createFormattedDescription () {
    return sharedEntity.processDescription(
      this.entity.description,
      this.entity.descriptionCredit
    )
  }

  get images () {
    return this.entity.images
  }

  get description () {
    return this.entity.description
  }

  get descriptionCredit () {
    return this.entity.descriptionCredit
  }

  get weSay () {
    return this.entity.weSay
  }

  get email () {
    return this.entity.email
  }

  get telephone () {
    return this.entity.telephone
  }

  get wheelchairAccessType () {
    return this.entity.wheelchairAccessType
  }

  get disabledBathroomType () {
    return this.entity.disabledBathroomType
  }

  get hearingFacilitiesType () {
    return this.entity.hearingFacilitiesType
  }

  get hasPermanentCollection () {
    return this.entity.hasPermanentCollection
  }

  get isFullEntity () {
    return true
  }

  shallowClone (newProps) {
    const clonedEntity = Object.assign({}, this.entity, newProps || {})
    return new FullVenue(clonedEntity)
  }

  getLinkByType (linkType) {
    return this.links.getLinkByType(linkType)
  }

  getHomepageUrl () {
    const homepage = this.getLinkByType(LINK_TYPE_HOMEPAGE)
    return homepage ? homepage.url : null
  }
}
