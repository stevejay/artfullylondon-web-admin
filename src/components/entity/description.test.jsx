import React from 'react'
import { shallow } from 'enzyme'

import { FullVenue } from '_src/entities/venue'
import EntityDescription from '_src/components/entity/description'

it('should render correctly', () => {
  const mockEntity = new FullVenue({})
  mockEntity.getLinkByType = jest.fn().mockReturnValue('https://wikipedia/link')

  mockEntity.createFormattedDescription = jest
    .fn()
    .mockReturnValue('Some description.')

  const wrapper = shallow(<EntityDescription entity={mockEntity} />)

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when there is no wikipedia link', () => {
  const mockEntity = new FullVenue({})
  mockEntity.getLinkByType = jest.fn().mockReturnValue(null)

  mockEntity.createFormattedDescription = jest
    .fn()
    .mockReturnValue('Some description.')

  const wrapper = shallow(<EntityDescription entity={mockEntity} />)

  expect(wrapper).toMatchSnapshot()
})

it('should not update', () => {
  const mockEntity = new FullVenue({})
  mockEntity.getLinkByType = jest.fn().mockReturnValue(null)
  mockEntity.createFormattedDescription = jest.fn().mockReturnValue('')

  const wrapper = shallow(<EntityDescription entity={mockEntity} />)

  const result = wrapper.instance().shouldComponentUpdate()
  expect(result).toEqual(false)
})
