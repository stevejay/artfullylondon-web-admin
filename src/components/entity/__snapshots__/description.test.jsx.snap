// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="src-components-entity-description__container"
>
  <div
    className="src-components-entity-description__html"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Some description.",
      }
    }
  />
  <a
    className="src-components-entity-description__link"
    rel="noopener"
    target="_blank"
  >
    Read more on Wikipedia
  </a>
</div>
`;

exports[`should render correctly when there is no wikipedia link 1`] = `
<div
  className="src-components-entity-description__container"
>
  <div
    className="src-components-entity-description__html"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Some description.",
      }
    }
  />
</div>
`;
